lib/engine-addon.js

```diff
@@ -488,8 +488,9 @@ module.exports = {
       };
 
       this.treeForEngine = function() {
-        // If this engine is lazy or any of its parents are lazy we need to promote its routes.
-        if (!this._hasLazyAncestor) { return; }
+        // Unless includeRoutesInApplication is false, if this engine or any of
+        // its parents are lazy we need to promote its routes.
+        if (!this._hasLazyAncestor || this.lazyLoading.includeRoutesInApplication === false) { return; }
```

Comment:
This line and the comment above it are hard to grok. I think this is likely because there isn't clear indication upfront about what this function does (and not your fault).

My suggestion: we should add a block comment to the function stating that `treeForEngine` returns a tree representing `routes.js` and its dependency graph.

Then, we can make this block more understandable by using a descriptive variable name:

```js
var extractRoutes = this._hasLazyAncestor && this.lazyLoad.includeRoutesInApplication !== false;
if (!extractRoutes) { return; }
```